### ISSUE TITLE
[Discussion] notify Subscribe/Unsubscribe subscriber when subscription is actually added

### ISSUE THREAD

**Issue by aav** [riker-rs/riker#77]
Maybe it makes sense to inform channel subscriber, that subscription has been added to the subscription list? 

Such notification can be useful in the case when subscriber has to be sure that it will be notified on something, it immediately initiated. This will help to avoid race conditions similar to what currently happens in _ActorSystem::shutdown._ I believe that this is quite a common pattern.

**Comment by leenozara** [comment]
@aav Yes, absolutely. This is a common pattern and should be implemented.

(Sorry for the very delayed response. See #82)
